src/Persistence_representations/include/gudhi/PSSK.h

```diff
@@ -5,163 +5,168 @@
-       size_t number_of_pixels = 1000,
-       double min_ = -1,
-       double max_ = -1,
-       unsigned dimension = std::numeric_limits<unsigned>::max())
-      : Persistence_heat_maps()
+       const std::vector<std::vector<double> >& filter = create_Gaussian_filter(5, 1),
+       std::size_t number_of_pixels = 1000,
+       double min = -1,
+       double max = -1,
+       unsigned int dimension = std::numeric_limits<unsigned int>::max())
+      : Base()
   {
-    std::vector<std::pair<double, double> > intervals_;
-    if (dimension == std::numeric_limits<unsigned>::max()) {
-      intervals_ = read_persistence_intervals_in_one_dimension_from_file(filename);
+    std::vector<std::pair<double, double> > intervals;
+    if (dimension == std::numeric_limits<unsigned int>::max()) {
+      intervals = read_persistence_intervals_in_one_dimension_from_file(filename);
     } else {
-      intervals_ = read_persistence_intervals_in_one_dimension_from_file(filename, dimension);
+      intervals = read_persistence_intervals_in_one_dimension_from_file(filename, dimension);
     }
-    this->construct(intervals_, filter, number_of_pixels, min_, max_);
+    this->_construct(intervals, filter, number_of_pixels, min, max);
   }
 
- protected:
-  void construct(const std::vector<std::pair<double, double> >& intervals_,
-                 std::vector<std::vector<double> > filter = create_Gaussian_filter(5, 1),
-                 size_t number_of_pixels = 1000,
-                 double min_ = -1,
-                 double max_ = -1);
-};
-
-// if min_ == max_, then the program is requested to set up the values itself based on persistence intervals
-void PSSK::construct(const std::vector<std::pair<double, double> >& intervals_,
-                     std::vector<std::vector<double> > filter,
-                     size_t number_of_pixels,
-                     double min_,
-                     double max_)
-{
-  bool dbg = false;
-  if (dbg) {
-    std::cerr << "Entering construct procedure \n";
-    getchar();
-  }
-
-  if (min_ == max_) {
-    // in this case, we want the program to set up the min_ and max_ values by itself.
-    min_ = std::numeric_limits<int>::max();
-    max_ = -std::numeric_limits<int>::max();
+ private:
+  // if min == max, then the program is requested to set up the values itself based on persistence intervals
+  void _construct(const std::vector<std::pair<double, double> >& intervals,
+                  const std::vector<std::vector<double> >& filter = create_Gaussian_filter(5, 1),
+                  std::size_t number_of_pixels = 1000,
+                  double min = -1,
+                  double max = -1)
+  {
+#ifdef GUDHI_DEBUG
+    std::clog << "Entering construct procedure \n";
+#endif
+
+    if (min == max) {
+      // in this case, we want the program to set up the min and max values by itself.
+      min = std::numeric_limits<double>::max();
+      max = std::numeric_limits<double>::min();
```

Comment:
If we want to handle negative numbers, `numeric_limits<double>::min` is wrong, it should be `lowest` (or `-max`).

Reply:
Ah, I never realized that `min` did not work the same for `int` and `double`/`float`. Weird choice, but I guess there must be a reason for it. I change it to `-max`.